service/matching/matchingEngine.go

```diff
@@ -115,6 +122,10 @@ type (
 		timeSource           clock.TimeSource
 		// Only set if global namespaces are enabled on the cluster.
 		namespaceReplicationQueue persistence.NamespaceReplicationQueue
+		// Disables concurrent task queue user data updates and replication requests (due to a cassandra limitation)
+		namespaceUpdateLockMap map[string]*namespaceUpdateLocks
+		// Serializes access to the per namespace lock map
+		namespaceUpdateLockMapLock sync.Mutex
```

Comment:
Not that I have any alternative but man, lockfest 2000

Reply:
Yeah, the future is here

Reply:
`IDMutex` is at least a little cleaner than this